src/workspace.ts

```diff
@@ -26,23 +26,25 @@ export type Packages = Dict<{
  * and the package JSONs themselves.
  */
 export function listPkgs(wsRoot: string, globs: string[]) {
-  let filesAndDirs = flatMap(globs, g => glob.sync(g))
+  // based on yarn v1.18.0 workspace resolution: https://github.com/yarnpkg/yarn/blob/v1.18.0/src/config.js#L794
+  const registryFilenames = ['package.json', 'yarn.json']
+  const registryFolders = ['node_modules']
+  const trailingPattern = `/+(${registryFilenames.join('|')})`
+  const ignorePatterns = registryFolders.map(folder => `/${folder}/**/+(${registryFilenames.join('|')})`)
```

Comment:
This does mimic yarn exactly, but there is a bug because it will also match files named `packageTjson`. Which... not a big deal, but might be useful to add a comment about this.